master/internal/api_generic_tasks.go

```diff
@@ -273,3 +274,35 @@ func (a *apiServer) CreateGenericTask(
 		Warnings: pkgCommand.LaunchWarningToProto(warnings),
 	}, nil
 }
+
+func (a *apiServer) PropegateTaskState(ctx context.Context, taskID model.TaskID, state model.TaskState) error {
```

Comment:
maybe i'm reviewing too early, but where/how do we plan on using this?

Reply:
When we kill or pause a task, the state will need to propagate to the child tasks

Reply:
We should probably chat a bit about how we plan to do this. I was thinking each task would be in charge of its own state, not that the parent would write them all. Either probably works, but curious to discuss the plan and how to works with restarts / fault tolerance.